### PR TITLE
Linter updates

### PR DESCRIPTION
**Pull request by lalli-flores**
This PR contains several changes related to linter and related tooling, notably:
* Updates remaining tooling versions
* Moves ESLint config to a JSON file
* Removes unused dev dependencies
* Updates ESLint and ESLint plugins to match the team's agreed styles
* Contains ESLint style overrides when team style deviates from Airbnb's style

### CODE REVIEW

**Comment by lalli-flores** [comment]
Good catch, @brunogenaro!
Fixed the typo in the commit message and updated the rules after the team discussion earlier today.

File: src/MeetingsAdapter.js [hunk 1]
```diff
@@ -3,61 +3,65 @@ import {throwError} from 'rxjs';
...
  */
 
 /**
  * A control/action that may modify a meeting or meeting state.
  *
- * @typedef {Object}  MeetingControl
- * @property {string}                         ID       The meeting control ID or name.
- * @property {MeetingControlAction}           action   A function that performs the control action. Must not return a value, as it must perform side-effects only.
- * @property {MeetingControlDisplayFunction}  display  A function that returns an observable that emits MeetingControlDisplay objects.
+ * @typedef {object}  MeetingControl
```
Comment (adamweeks) [inline src/MeetingsAdapter.js:36]: I really need a linter that will tell me the proper capitalization for `string` and `object` in jsdoc.

Comment (lalli-flores) [reply src/MeetingsAdapter.js:36]: Behold, the ESLint JSDoc plugin does! 😁
Hence all the changes

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    ],
+    "no-dupe-keys": "error",
+    "no-dupe-class-members": "error",
+    "newline-after-var": "error",
+    "newline-before-return": "error",
+    "lines-around-directive": "error",
+    "no-useless-call": "error"
+  },
+  "overrides": [
```
Comment (adamweeks) [inline .eslintrc.json:73]: Is there a way to make these overrides a separate file so that the eslint.json file can be shared throughout repos?

Comment (lalli-flores) [reply .eslintrc.json:73]: Yes, my plan was just to get this through the component system repos to test pure JS and JS/JSX + React. Once that was finished then I was gonna refactor it out into the web styles repo and just leave the overrides here.

**Pull request merged**